AdyenEncryption/CardEncryptor.swift

```diff
@@ -37,161 +21,211 @@ public enum CardEncryptor {
     ///  maybe because the card public key is an invalid one, or for any other reason.
     /// - Throws:  `CardEncryptor.Error.invalidEncryptionArguments` when trying to encrypt a card with  card number, securityCode,
     /// expiryMonth, and expiryYear, all of them are nil.
-    public static func encryptedCard(for card: Card, publicKey: String) throws -> EncryptedCard {
-        guard !card.isEmpty else {
-            throw CardEncryptor.Error.invalidEncryptionArguments
+    public static func encrypt(card: Card, with publicKey: String) throws -> EncryptedCard {
+        guard
+            let number = card.number,
+            let securityCode = card.securityCode,
+            let expiryMonth = card.expiryMonth,
+            let expiryYear = card.expiryYear
+        else { throw Error.invalidCard }
+
+        let builder = CardPayloadBuilder()
+        return EncryptedCard(number: try encrypt(builder.add(number: number), with: publicKey),
+                             securityCode: try encrypt(builder.add(securityCode: securityCode), with: publicKey),
+                             expiryMonth: try encrypt(builder.add(expiryMonth: expiryMonth), with: publicKey),
+                             expiryYear: try encrypt(builder.add(expiryYear: expiryYear), with: publicKey))
```

Comment:
this is really weird, so the first line encrypts the payload with only number, the second line encrypts the payload with number and securityCode, the third line encrypts payload with number, securityCode and expiryMonth?.